nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessAnalysis.java

```diff
@@ -211,12 +212,25 @@ public NullnessStore getNullnessInfoBeforeNestedMethodNode(
     return store.filterAccessPaths(
         (ap) -> {
           boolean allAPNonRootElementsAreFinalFields = true;
-          for (AccessPathElement ape : ap.getElements()) {
+          ImmutableList<AccessPathElement> elements = ap.getElements();
+          for (int i = 0; i < elements.size(); i++) {
+            AccessPathElement ape = elements.get(i);
             Element e = ape.getJavaElement();
-            if (!e.getKind().equals(ElementKind.FIELD)
-                || !e.getModifiers().contains(Modifier.FINAL)) {
-              allAPNonRootElementsAreFinalFields = false;
-              break;
+            if (i != elements.size() - 1) { // "inner" elements of the access path, must be fields
```

Comment:
Why should inner elements of the access path be fields? And why couldn't we keep the original "e.getKind().equals(ElementKind.FIELD)" check?

Reply:
I restored the checks for fields and added tests showing why they are required in bbfd36ce64ff61f8af556a53150446e3cf3a438c